lib/sober_swag/nodes/binary.rb

```diff
@@ -1,37 +1,43 @@
 module SoberSwag
   module Nodes
     ##
-    # A
-    #
-    # It's cool I promise.
+    # A binary node: has a left and right hand side.
+    # Basically a node of a binary tree.
     class Binary < Base
+      ##
+      # @param lhs [SoberSwag::Nodes::Base] the left-hand node.
+      # @param rhs [SoberSwag::Nodes::Base] the right-hand node.
       def initialize(lhs, rhs)
         @lhs = lhs
         @rhs = rhs
       end
 
-      attr_reader :lhs, :rhs
+      ##
+      # @return [SoberSwag::Nodes::Base] the left-hand node
+      attr_reader :lhs
 
       ##
-      # Map the root values of the node.
-      # This just calls map on the lhs and the rhs
-      def map(&block)
-        self.class.new(
-          lhs.map(&block),
-          rhs.map(&block)
-        )
-      end
+      # @return [SoberSwag::Nodes::Base] the right-hand node
```

Comment:
Would it be useful to add aliases for left/right?
- left_child
- left_node
etc?